src/cascadia/TerminalSettingsEditor/Profiles.cpp

```diff
@@ -15,16 +15,71 @@ using namespace winrt::Windows::UI::Xaml::Data;
 using namespace winrt::Windows::UI::Xaml::Navigation;
 using namespace winrt::Windows::Foundation;
 using namespace winrt::Windows::Foundation::Collections;
-using namespace winrt::Windows::Storage;
-using namespace winrt::Windows::Storage::AccessCache;
-using namespace winrt::Windows::Storage::Pickers;
 using namespace winrt::Microsoft::Terminal::Settings::Model;
 
 static const std::array<winrt::guid, 2> InBoxProfileGuids{
     winrt::guid{ 0x61c54bbd, 0xc2c6, 0x5271, { 0x96, 0xe7, 0x00, 0x9a, 0x87, 0xff, 0x44, 0xbf } }, // Windows Powershell
     winrt::guid{ 0x0caa0dad, 0x35be, 0x5f56, { 0xa8, 0xff, 0xaf, 0xce, 0xee, 0xaa, 0x61, 0x01 } } // Command Prompt
 };
 
+// Function Description:
+// - This function presents a File Open "common dialog" and returns its selected file asynchronously.
+// Parameters:
+// - customize: A lambda that receives an IFileDialog* to customize.
+// Return value:
+// (async) path to the selected item.
+template<typename TLambda>
+static winrt::Windows::Foundation::IAsyncOperation<winrt::hstring> OpenFilePicker(TLambda&& customize)
+{
+    auto fileDialog{ winrt::create_instance<IFileDialog>(CLSID_FileOpenDialog) };
+    DWORD flags{};
+    THROW_IF_FAILED(fileDialog->GetOptions(&flags));
+    THROW_IF_FAILED(fileDialog->SetOptions(flags | FOS_FORCEFILESYSTEM | FOS_NOCHANGEDIR | FOS_DONTADDTORECENT)); // filesystem objects only; no recent places
+    customize(fileDialog.get());
+
+    auto hr{ fileDialog->Show(NULL) };
+    if (!SUCCEEDED(hr))
+    {
+        if (hr == HRESULT_FROM_WIN32(ERROR_CANCELLED))
+        {
+            co_return winrt::hstring{};
+        }
+        THROW_HR(hr);
+    }
+
+    winrt::com_ptr<IShellItem> result;
+    THROW_IF_FAILED(fileDialog->GetResult(result.put()));
+
+    wil::unique_cotaskmem_string filePath;
+    THROW_IF_FAILED(result->GetDisplayName(SIGDN_FILESYSPATH, &filePath));
+
+    co_return winrt::hstring{ filePath.get() };
+}
+
+// Function Description:
+// - Helper that opens a file picker pre-seeded with image file types.
+static winrt::Windows::Foundation::IAsyncOperation<winrt::hstring> OpenImagePicker()
+{
+    static constexpr COMDLG_FILTERSPEC supportedImageFileTypes[] = {
+        { L"All Supported Bitmap Types (*.jpg, *.jpeg, *.png, *.bmp, *.gif, *.tiff, *.ico)", L"*.jpg;*.jpeg;*.png;*.bmp;*.gif;*.tiff;*.ico" },
+        { L"All Files (*.*)", L"*.*" }
+    };
+
+    static constexpr winrt::guid clientGuidImagePicker{ 0x55675F54, 0x74A1, 0x4552, { 0xA3, 0x9D, 0x94, 0xAE, 0x85, 0xD8, 0xF2, 0x7A } };
+    return OpenFilePicker([](auto&& dialog) {
+        THROW_IF_FAILED(dialog->SetClientGuid(clientGuidImagePicker));
```

Comment:
the client GUID is how it remembers the location, so we can have it remember different locations for different types of box!